test/integration/fuzzing_test.lua

```diff
@@ -0,0 +1,2673 @@
+                list_type = types.list(types.nonNull(graphql_types[argument[ARGUMENT_INNER_TYPE]].graphql_type))
+            else
+                list_type = types.list(graphql_types[argument[ARGUMENT_INNER_TYPE]].graphql_type)
+            end
+            if argument[ARGUMENT_NULLABILITY] == NonNullable then
+                list_type = types.nonNull(list_type)
+            end
+        else
+            if #argument > 1 and argument[ARGUMENT_NULLABILITY] == NonNullable then
+                list_type = types.nonNull(graphql_types[argument[ARGUMENT_TYPE]].graphql_type)
+            else
+                list_type = graphql_types[argument[ARGUMENT_TYPE]].graphql_type
+            end
+        end
+        return list_type
+    end
+
+    local function gen_fields(_argument_list)
+        local args = {}
+        for k, v in ipairs(_argument_list) do
+            if v[ARGUMENT_TYPE] == 'inputObject' then
+                args['arg'..tostring(k)] = my_object
+            elseif v[ARGUMENT_TYPE] == 'list' then
+                if v[ARGUMENT_INNER_TYPE] == 'inputObject' then
+                    args['arg'..tostring(k)] = types.list(my_object)
+                else
+                    args['arg'..tostring(k)] = make_type(v)
+                end
+            else
+                args['arg'..tostring(k)] = make_type(v)
+            end
+        end
+        return args
+    end
+
+    local function gen_arguments(_argument_list)
+        local args = {}
+        for k, v in ipairs(_argument_list) do
+            args['arg'..tostring(k)] = make_type(v)
+        end
+        return args
+    end
+
+    return {
+        ['test'] = {
+            kind = types.object({
+                name = 'result',
+                fields = gen_fields(argument_list)
+            }),
+            arguments = gen_arguments(argument_list),
+            resolve = function(_, args)
+                return args
+            end,
+        }
+    }
+end
+
+local function gen_value(argument_type, argument_value)
+    local value = ''
+    if argument_value == nil then
+        value = 'null'
+    elseif argument_type == 'string' or argument_type == 'id' or
+            argument_type == 'custom_string' then
+        value = '"'..tostring(argument_value)..'"'
+    elseif argument_type == 'custom_json' then
+        value = '"""'..tostring(argument_value)..'"""'
+    elseif argument_type == 'inputObject' then
+        local fields_count = 0
+        for k1, v1 in pairs(argument_value) do
+            fields_count = fields_count + 1
+            value = value..'{ '..k1..': "'..tostring(v1)..'"'
+            if fields_count < #argument_value then
+                value = value..', '
+            end
+        end
+        if value ~= '' then
+            value = value..' }'
+        end
+    else
+        value = tostring(argument_value)
+    end
+    return value
+end
+
+local function gen_query(argument_list)
```

Comment:
I suggest to split this function to several small functions. It's too complex, it's impossible to understand that it does

Reply:
Turn on debug from previous comment and you will see the result of query generation based on provided arguments attributes

Reply:
Added comments to gen_query because dividing it to separate functions will not make it more understandable